components/calendar/src/astronomy.rs

```diff
@@ -457,6 +876,67 @@ impl Astronomical {
         .1
     }
 
+    #[allow(dead_code)]
+    fn moonset(date: Moment, location: Location) -> Option<Moment> {
+        let moment = Location::universal_from_standard(date, location);
+        let waxing = Self::lunar_phase(date) < 180.0;
+        let alt = Self::observed_lunar_altitude(moment, location);
+        let lat = location.latitude;
+        let offset = alt / (4.0 * (90.0 - libm::fabs(lat)));
+
+        let approx = if waxing {
+            if offset > 0.0 {
+                moment + offset
+            } else {
+                moment + 1.0 + offset
+            }
+        } else {
+            moment - offset + 0.5
+        };
+
+        let set = Moment::new(binary_search(
+            approx.inner() - (6.0 / 24.0),
+            approx.inner() + (6.0 / 24.0),
+            |x| Self::observed_lunar_altitude(Moment::new(x), location) < 0.0,
+            |u, l| (u - l) < 1.0 / (24.0 * 60.0),
+        ));
+
+        if set < moment + 1.0 {
+            let std = Moment::new(libm::fmax(
+                Location::standard_from_universal(set, location).inner(),
+                date.inner(),
+            ));
+            if std < date {
```

Comment:
use a `debug_assert!` because this condition shouldn't be possible based on the previous line's assignment involving `fmax` taking `date` as an arg